### ISSUE TITLE
How to calculate the dice？ 

### ISSUE THREAD

**Issue by shihe407** [Luo-Yihao/GHDHeart#3]
Thanks for your code. I have met some question and I hope you can tell me.
How do you calculate the dice of the left ventricle, and why doesn't the code given in the demo code need to add point_list[3]? In my understanding, the part of points_list[3] is the myocardial part, which is not needed for dice calculation?
`final_dice = 1 - paraheart.dice_evaluation(points_lv, torch.cat([point_list[0], point_list[1]], dim=0))
print('Final Dice: %.4f'%(final_dice*100)+'%')`
Also, if I need to calculate the dice of the right ventricle, do I need to select point_list[0], point_list[2], and point_list[3] to add to the dice calculation?

**Comment by Luo-Yihao** [comment]
Hi, when calculating a binary loss, we only focus on "points_lv" and  "points_out_of_lv." In this case, point_list[2] is myocardial, and cat[point_list[0], point_list[1]] is its complementary, if my memory is correct. One can calculate the dice for any part by sending itself and its complementary set into our .dice_evaluation() function. Thanks

**Comment by shihe407** [comment]
![image](https://github.com/user-attachments/assets/af3402d8-f361-4dcd-8e5e-135a3f0b2489)

Sorry I'm still a bit confused.
When visualising the point cloud, the code looks like `color_list = ['lightblue', 'lightsalmon', 'lightgreen']
for point, color in zip(point_list[1:], color_list):
    pl.add_points(point.cpu().numpy(), color=color, point_size=10, opacity=0.2),` my understanding should be that point_list[0] is the background part, point_list[1] light blue is supposed to be the right ventricle, point_list[2] is the left ventricle and point_list[3] light green is the blood pool of the left ventricle as I understand it.
I don't really understand the calculation of binary dice, can you explain more? Or tell me the resources I should consult to learn it.

**Comment by Luo-Yihao** [comment]
Great~ When u try to calculate the dice for Lv myocardial, consider 1 - paraheart.dice_evaluation(point_list[2], torch.cat([point_list[0], point_list[1], point_list[3]], dim=0)). Sorry for missing the cavity part. We will fix it later

**Comment by shihe407** [comment]
Thanks for your prompt reply and looking forward to the new release!
